docs/dev/clusteroperator.md

```diff
@@ -76,7 +76,20 @@ Here are the guarantees components can get when they follow the rules we define:
 There are a set of guarantees components are expected to honor in return:
 
 1. A component doesn't report the `Available` status condition the first time until they are completely rolled out (or within some reasonable percentage if the component must be installed to all nodes)
-2. A component reports `Failing` when it can't accomplish its task. This might be very broad - the API servers are down so I failed to update - or narrow - I couldn't update the last secret. In either case, Failing communicates something is wrong.
+2. A component reports `Degraded` when it can't accomplish its task or if its
+   current state does not match its desired state.  For example, if you desire 3
+   pods, and 2 are running, and 1 is crashlooping, your operator service is
+   possibly still `Available`, but its service is in a `Degraded` state.  A
+   component may be `Degraded` while it is `Progressing` to a new desired state;
+   for example, if only 2 of the 3 desired replicas are achieved.  As a result,
+   it may be normal for a operator during upgrade to temporarily report
+   `Degraded`.  A component may be `Progressing` but not `Degraded` if a config
+   change is happening that does not require a roll-out of new pods.  If your
+   rolling deployment bursts above its target replica count, you may be
+   `Progressing` but not `Degraded` because you have your desired number of pods
+   running to meet your service demand.  If  a component remains degraded for an
```

Comment:
I would think another indicator of a persistent Degraded state is Degraded=true and Progressing=false.

I still wish there was a way to indicate the notion of "external intervention is needed to get out of Degraded=true"